asab/metrics/service.py

```diff
@@ -120,7 +120,37 @@ def _add_metric(self, metric: Metric, metric_name: str, tags=None, reset=None, h
 
 		# Add local static tags
 		if tags is not None:
-			metric.StaticTags.update(tags)
+			tags_to_update = {}
```

Comment:
Tag key and value must be a string. You should do only the value check here and fix the code that tries to use other types as tags.

Reply:
Metrics can be initialized in runtime. If the check raises error for non-string tags it will end up in crash during runtime if the tags are not handled properly. Should we be that cruel and raise error or handle the situation (like omit the malicious tag from the StaticTags?)

I would raise ValueError or AssertionError here and look for kind solution inside the microservice as it can be different for different scenarios, but it is cruel...
@ateska

Reply:
OK